test/integration/scheduler/workload_controller_test.go

```diff
@@ -467,6 +470,205 @@ var _ = ginkgo.Describe("Workload controller with scheduler", func() {
 		})
 	})
 
+	ginkgo.When("Resource transformations are applied", func() {
+		ginkgo.BeforeEach(func() {
+			gomega.Expect(k8sClient.Create(ctx, onDemandFlavor)).To(gomega.Succeed())
+			clusterQueue = testing.MakeClusterQueue("clusterqueue").
+				ResourceGroup(*testing.MakeFlavorQuotas(onDemandFlavor.Name).
+					Resource(corev1.ResourceCPU, "5", "5").Obj()).
+				Cohort("cohort").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).To(gomega.Succeed())
+			localQueue = testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+		})
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, clusterQueue, true)
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, onDemandFlavor, true)
+		})
+
+		ginkgo.It("The transformed resources should be used as request values", func() {
+			features.SetFeatureGateDuringTest(ginkgo.GinkgoTB(), features.ConfigurableResourceTransformations, true)
+			features.SetFeatureGateDuringTest(ginkgo.GinkgoTB(), features.WorkloadResourceRequestsSummary, true)
+			var wl2 *kueue.Workload
+			ginkgo.By("Create and wait for workload admission", func() {
+				gomega.Expect(k8sClient.Create(ctx, localQueue)).To(gomega.Succeed())
+				wl = testing.MakeWorkload("one", ns.Name).
+					Queue(localQueue.Name).
+					Request(pseudoCPU, "1").
+					Obj()
+				gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+				util.ExpectWorkloadsToHaveQuotaReservation(ctx, k8sClient, clusterQueue.Name, wl)
+			})
+
+			ginkgo.By("Check queue resource consumption", func() {
+				gomega.Eventually(func(g gomega.Gomega) kueue.ClusterQueueStatus {
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterQueue), &updatedCQ)).To(gomega.Succeed())
+					return updatedCQ.Status
+				}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(kueue.ClusterQueueStatus{
+					PendingWorkloads:   0,
+					ReservingWorkloads: 1,
+					FlavorsReservation: []kueue.FlavorUsage{{
+						Name: kueue.ResourceFlavorReference(onDemandFlavor.Name),
+						Resources: []kueue.ResourceUsage{{
+							Name:  corev1.ResourceCPU,
+							Total: resource.MustParse("2"), // conversionBaseFactor is 2
+						}},
+					}},
+				}, ignoreCqCondition, ignoreInClusterQueueStatus))
+			})
+
+			ginkgo.By("Check podSets spec", func() {
+				wlRead := kueue.Workload{}
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), &wlRead)).To(gomega.Succeed())
+				gomega.Expect(equality.Semantic.DeepEqual(wl.Spec.PodSets, wlRead.Spec.PodSets)).To(gomega.BeTrue())
```

Comment:
@dgrove-oss Could you open a follow-up PR to switch this `semantic.DeepEqual` to the `gomega.BeComparableTo`?